### PR TITLE
Feature/circle ci  init

### PR DESCRIPTION
**Pull request by kevinahuber**
Adds Circle CI to [pysonar](https://circleci.com/gh/OpenMined/PySonar), and adds pep8 style check.

### CODE REVIEW

File: sonar/contracts.py [hunk 1]
```diff
@@ -142,36 +149,41 @@ def get_transaction(self,from_addr,value=None):
...
-
-        """
-
         ipfs_address = self.ipfs.add_pyobj(model.syft_obj)
-        deploy_trans = self.get_transaction(model.owner,value=self.web3.toWei(model.bounty,'ether')).addModel([ipfs_address[0:32],ipfs_address[32:]],model.initial_error,model.target_error)
-        return self.call.getNumModels()-1
+        deploy_trans = self.get_transaction(model.owner,
+        value=self.web3.toWei(model.bounty,'ether'))
+        .addModel([ipfs_address[0:32], ipfs_address[32:]], model.initial_error,
+        model.target_error)"""
```
Comment (kevinahuber) [inline sonar/contracts.py]: These parts weren't used, are they supposed to be?

**Comment by kevinahuber** [comment]
This took wayyyy too long (16 min)- I'm going to look into ways to speed it up

**Comment by anoff** [comment]
> This took wayyyy too long (16 min)- I'm going to look into ways to speed it up

Building docker images always takes ages due to `numpy` and `scipy` depdendencies. One way out might be automatic builds via dockerhub.

**Comment by kevinahuber** [comment]
@anoff toyed around with it, but I'm not seeing a clear path. Automated build could fulfill the deploy portion of `circle-ci`, but we still need to run tests, which need `numpy` and `scipy`, right?

**Comment by kevinahuber** [comment]
We could build a base image with `numpy` and `scipy` first, then build on that.

**Comment by kevinahuber** [comment]
down to 24s. Feels a bit hacky though.

**Pull request closed without merging**